### DISCUSSION TITLE
Composer: unsupported value if != constraint used

### DISCUSSION THREAD

**Discussion by martipoe** [renovatebot/renovate#31501] [Request Help]
### What would you like help with?

I think I found a bug

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitlab

### Please tell us more about your question or problem

I am using this version constraint with PHP Composer in composer.json:
`"typo3/cms-core": "^12.4.7 !=12.4.8",`

This syntax should be supported as per https://semver.madewithlove.com/?package=typo3%2Fcms-core&constraint=%5E12.4.7+!%3D12.4.8&stability=stable

However, RenovateBot seems not to support this syntax (see logs). 

Consequently, this package and all packages depending on it are not updated.

It would be nice if this could be fixed/supported.

### Logs (if relevant)

<details><summary>Logs</summary>

```
DEBUG: Dependency typo3/cms-core has unsupported/unversioned value ^12.4.7 !=12.4.8 (versioning=composer)`
```

</details>

**Comment by rarkins** [comment]
It seems like it's the `!=` part causing the problem. Here's some example tests I added:

```
    ✓ isValid("^12.4.7") === true
    ✕ isValid("!=12.4.8") === true (1 ms)
    ✕ isValid("^12.4.7 !=12.4.8") === true
    ✓ isValid("^12.4.7 >12.4.8") === true
    ✕ isValid("^12.4.7 | !=12.4.8") === true (1 ms)
    ✕ isValid("^12.4.7 || !=12.4.8") === true
```

**Comment by rarkins** [comment]
Converted to issue: https://github.com/renovatebot/renovate/issues/31504
